### PR TITLE
PVF: Fix external workers being required for non-collator full nodes

### PR DESCRIPTION
**Pull request by mrcnski**
See title and [this thread](https://github.com/paritytech/cumulus/pull/2929#issuecomment-1658537276).

Cumulus companion: https://github.com/paritytech/cumulus/pull/2929

### CODE REVIEW

File: cli/src/cli.rs [hunk 1]
```diff
@@ -71,17 +71,6 @@ pub enum Subcommand {
...
-#[allow(missing_docs)]
-#[derive(Debug, Parser)]
-pub struct ValidationWorkerCommand {
-	/// The path to the validation host's socket.
-	#[arg(long)]
-	pub socket_path: String,
-	/// Calling node implementation version
-	#[arg(long)]
-	pub node_impl_version: String,
-}
```
Comment (mrcnski) [inline cli/src/cli.rs:83]: Sorry for unrelated change, this was dead code

File: node/service/src/lib.rs [hunk 2]
```diff
@@ -909,7 +909,7 @@ pub fn new_full<OverseerGenerator: OverseerGen>(
 		slot_duration_millis: slot_duration.as_millis() as u64,
 	};
 
-	let candidate_validation_config = if is_collator.is_collator() {
+	let candidate_validation_config = if role.is_authority() && !is_collator.is_collator() {
```
Comment (rphmeier) [inline node/service/src/lib.rs:912]: doesn't this instantiate a dummy subsystem for candidate validation for every node that's an authority and is not a collator?

Comment (mrcnski) [reply node/service/src/lib.rs:912]: Sorry, did that too fast!

Wanted to keep the conditional as `role.is_authority() && !is_collator.is_collator()` instead of negating it, since it's the same conditional that's used for `pre_checker_enabled` above.

Comment (rphmeier) [reply node/service/src/lib.rs:912]: This might as well be `if role.is_authority()` as there aren't really any nodes that are both authorities and collators. But if there were a node that is an authority and a collator, it'd probably want to validate candidates (in its role as an authority).

Comment (rphmeier) [reply node/service/src/lib.rs:912]: The check above is probably wrong too, based on first principles.

Comment (rphmeier) [reply node/service/src/lib.rs:912]: This doesn't capture the possibility of nodes not being authorities and also checking parablocks, but we can cross that bridge when we get to it. Currently dispute infrastructure doesn't allow it, so kind of a moot point. I do wish these interfaces were a bit more forward thinking...

Comment (mrcnski) [reply node/service/src/lib.rs]: > The check above is probably wrong too, based on first principles.

I'm kind of scared to touch it here. Changed the current check though.

**Review by skunert — APPROVED** [review]
LGTM

**Review by rphmeier — APPROVED** [review]
looks good as a quick fix

**Comment by mrcnski** [comment]
bot merge

File: node/service/src/lib.rs (hunk 2 above)
Comment (mrcnski) [reply node/service/src/lib.rs:912]: @rphmeier The zombienet tests were failing without the `!is_collator` check so I guess it is needed. (local cumulus tests had worked without it though.) Perhaps this is something I can revisit in the refactor, for now I will merge this in to fix CI.

**Comment by mrcnski** [comment]
bot merge
